### PR TITLE
ARROW-6250: [Java] Implement ApproxEqualsVisitor comparing approx for floating point

### PR DESCRIPTION
**Pull request by tianchen92**
Related to [ARROW-6250](https://issues.apache.org/jira/browse/ARROW-6250).

Currently we already implemented RangeEqualsVisitor/VectorEqualsVisitor for comparing range/vector.
And ARROW-6211 is created to make ValueVector work with generic visitor.
We should also implement ApproxEqualsVisitor to compare floating point just like cpp does
https://github.com/apache/arrow/blob/master/cpp/src/arrow/compare.cc

### CODE REVIEW

**Comment by tianchen92** [comment]
cc @pravindra

File: java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Visitor to compare floating point.
+ */
+public class ApproxEqualsVisitor extends VectorEqualsVisitor {
+
+  private final float epsilon;
+
+  public ApproxEqualsVisitor(ValueVector right, float epsilon) {
```
Comment (pravindra) [inline java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java]: can you pls add a comment for the param ? eg. the float/double values are treated as equal as long as the delta is <= epsilon.

File: java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  @Override
+  public Boolean visit(BaseFixedWidthVector left, Void value) {
+    if (left instanceof Float4Vector) {
+      return validate(left) && float4ApproxEquals((Float4Vector) left);
+    } else if (left instanceof Float8Vector) {
+      return validate(left) && float8ApproxEquals((Float8Vector) left);
+    } else {
+      return super.visit(left, value);
```
Comment (pravindra) [inline java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java]: doesn't work with the list of floats, struct/unions with floats ..

File: java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java (hunk 1 above)
Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java]: Thanks, fixed.

File: java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java (hunk 2 above)
Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java]: fixed and added some tests.

**Comment by tianchen92** [comment]
@pravindra I updated this PR, please help take a look when you have time, thanks!

**Review by pravindra — APPROVED** [review]

**Pull request closed without merging**